asset-transfer-basic/chaincode-java/src/main/java/org/hyperledger/fabric/samples/assetTransfer/AssetTransfer.java

```diff
@@ -0,0 +1,238 @@
+        }
+        
+        String assetState = stub.getStringState(assetID);
+        Asset asset = genson.deserialize(assetState, Asset.class);
+
+        Asset newAsset = new Asset(assetID, color, size, owner, appraisedValue);
+        String newAssetState = genson.serialize(newAsset);
+        stub.putStringState(assetID, newAssetState);
+
+        return newAsset;
+    }
+    
+    /**
+     * Deletes asset on the ledger.
+     *
+     * @param ctx the transaction context
+     * @param assetID the assetID
+     */
+    @Transaction()
+    public void DeleteAsset(final Context ctx, final String assetID) {
+        ChaincodeStub stub = ctx.getStub();
+
+        if (!AssetExists(ctx, assetID)) {
+            String errorMessage = String.format("Asset %s does not exist", assetID);
+            System.out.println(errorMessage);
+            throw new ChaincodeException(errorMessage, AssetTransferErrors.ASSET_NOT_FOUND.toString());
+        }
+
+        stub.delState(assetID);
+    }
+
+    /**
+     * Checks the existence of the asset on the ledger
+     *
+     * @param ctx the transaction context
+     * @param assetID the asset key
+     */
+    @Transaction()
+    public boolean AssetExists(final Context ctx, final String assetID) {
+        ChaincodeStub stub = ctx.getStub();
+        String assetState = stub.getStringState(assetID);
+
+        return (assetState != null && !assetState.isEmpty());
+
+    }
+
+    /**
+     * Changes the owner of a asset on the ledger.
+     *
+     * @param ctx the transaction context
+     * @param assetID the assetID
+     * @param newOwner the new owner
+     * @return the updated Asset
+     */
+    @Transaction()
+    public Asset TransferAsset(final Context ctx, final String assetID, final String newOwner) {
+        ChaincodeStub stub = ctx.getStub();
+
+        if (!AssetExists(ctx, assetID)) {
+            String errorMessage = String.format("Asset %s does not exist", assetID);
+            System.out.println(errorMessage);
+            throw new ChaincodeException(errorMessage, AssetTransferErrors.ASSET_NOT_FOUND.toString());
+        }
+        
+        String assetState = stub.getStringState(assetID);
+        Asset asset = genson.deserialize(assetState, Asset.class);
+
+        Asset newAsset = new Asset(asset.getAssetID(), asset.getColor(), asset.getSize(), newOwner, asset.getAppraisedValue());
+        String newAssetState = genson.serialize(newAsset);
+        stub.putStringState(assetID, newAssetState);
+
+        return newAsset;
+    }
+    
+    /**
+     * Retrieves all assets from the ledger.
+     *
+     * @param ctx the transaction context
+     * @return array of Assets found on the ledger
+     */
+    @Transaction()
+    public String GetAllAssets(final Context ctx) {
+        ChaincodeStub stub = ctx.getStub();
+
+        List<AssetQueryResult> queryResults = new ArrayList<AssetQueryResult>();
```

Comment:
if we go with GO chaincode,  getAll should return `ArrayList<Asset>`, so the class AssetQueryResult will get removed.

https://github.com/hyperledger/fabric-samples/blob/971af50cdfdffac9b7645db244d1c57474709ffd/asset-transfer-basic/chaincode-go/chaincode/smartcontract.go#L160

Reply:
So the function is returning ```ArrayList<Asset>``` instead of ```String``` ?

Reply:
Just tested this, and it results in an error when I return an ```ArrayList<Asset>```. I will get rid of the ```AssetQueryResult``` class and return ```final String response = genson.serialize(queryResults);``` where queryResults is an ```ArrayList<Asset>```.

Reply:
we want to return a json array of Assets `[ {asset1}, {asset2}]` like in go

is genson.serialize(arraylist of assets) working?

Reply:
Yes queryResults(asset arraylist) serializes into a json array of Assets without issue. It formats it into a string just like in the JS and TS chaincodes.